src/frontend/src/common/components/library/data_subview/components/CreateNSTreeModal/index.tsx

```diff
@@ -79,10 +84,32 @@ export const CreateNSTreeModal = ({
     []
   );
 
-  // Certain tree types can filter based on lineages and date ranges
+  // Certain tree types can filter based on lineages
   const { data: lineagesData } = useLineages();
   const availableLineages: string[] = lineagesData?.lineages || [];
   const [selectedLineages, setSelectedLineages] = useState<string[]>([]);
+
+  // Filter based on location
+  const { data: groupInfo } = useGroupInfo();
+  const { data: namedLocationsData } = useNamedLocations();
+  const namedLocations: NamedGisaidLocation[] =
+    namedLocationsData?.namedLocations ?? [];
+
+  // If we have the group's location, use this as the default for the filter
+  const [selectedLocation, setSelectedLocation] =
+    useState<NamedGisaidLocation | null>(
+      groupInfo?.location ? foldInLocationName(groupInfo?.location) : null
+    );
+
+  // If the group call isn't back when this is loaded, we need to update when the
+  // call returns
+  useEffect(() => {
+    setSelectedLocation(
+      groupInfo?.location ? foldInLocationName(groupInfo?.location) : null
+    );
+  }, [groupInfo?.location]);
+
```

Comment:
I'm not sure why other places in the app can use useGroup without putting it in a useEffect. 

This looks awkward, open to suggestions.

Reply:
Hmmm, I'm not sure what would be causing that. Is it that you set something off the groupInfo once it gets loaded? @mmmmmmaya might have a better idea

Reply:
I do set something off of group data once it's loaded, but I am under the impression this is also what happens in other components, which seem to work without this.  😕